server/src/main/java/org/elasticsearch/cluster/routing/IndexRouting.java

```diff
@@ -334,16 +334,13 @@ private void extractItem(String path, XContentParser source) throws IOException
                         source.nextToken();
                         break;
                     case VALUE_STRING:
+                    case VALUE_NUMBER:
                         hashes.add(new NameAndHash(new BytesRef(path), hash(new BytesRef(source.text()))));
```

Comment:
This isn't related to this change, but I think we should delegate to `addMatching(...)` here? Otherwise all fields are taken into account while hashing?